ema_workbench/em_framework/evaluators.py

```diff
@@ -415,6 +415,52 @@ def evaluate_experiments(self, scenarios, policies, callback, combine="factorial
         add_tasks(self.n_processes, self._pool, ex_gen, callback)
 
 
+class MPIEvaluator(BaseEvaluator):
+    """Evaluator for experiments using MPI Pool Executor from mpi4py"""
+
+    def __init__(self, msis, **kwargs):
+        from mpi4py.futures import MPIPoolExecutor
+
+        super().__init__(msis, **kwargs)
+        self._pool = None
+
+    def initialize(self):
+        self._pool = MPIPoolExecutor()
+        _logger.info(f"MPI pool started with {self._pool._max_workers} workers")
+        return self
+
+    def finalize(self):
+        self._pool.shutdown()
+        _logger.info("MPI pool has been shut down")
+
+    def evaluate_experiments(self, scenarios, policies, callback, combine="factorial"):
+        ex_gen = experiment_generator(scenarios, self._msis, policies, combine=combine)
+        experiments = list(ex_gen)  # Convert generator to list
+
+        # Create the model map just like in SequentialEvaluator
+        models = NamedObjectMap(AbstractModel)
+        models.extend(self._msis)
+
+        # Pack models with each experiment
+        packed = [(experiment, models) for experiment in experiments]
+
+        # Use the pool to execute in parallel
+        results = self._pool.map(run_experiment_mpi, packed)
+
+        for experiment, outcomes in results:
+            callback(experiment, outcomes)
+
+
+def run_experiment_mpi(packed_data):
```

Comment:
You now instantiate an ExperimentRunner for each separate experiment. In principle, you should have to instantiate this just once by making it a global. Not sure about the performance overhead of this. Related, you NWO  pack all models for each experiment, again this might not be the most efficient. Using an initializer function can solve both problems (see MultiprocessingEvaluator and ema_multiprocessing.py)

Reply:
I implemented the global ExperimentRunner in b8bf0e7. Is this what you had in mind?

Reply:
yes, this is now also consistent with how it is handled in multiprocessing